### DISCUSSION TITLE
Administrative task

### DISCUSSION THREAD

**Discussion by ChristiaanEmpowerBI** [MicroStrategy/rest-api-docs#45] [Q&A] [answered]
I am looking for the ability to get a list of reports, dossiers and documents, together with the last date that they were opened, and where they are placed within the folder structure of the server. 

Are there REST API calls that I can make to do this?

This can be done in Command Manager, so I am assuming it can be done in the REST API, but I am not finding method that can do this yet.

**Comment by ChristiaanEmpowerBI** [comment]
I was able to get this with the help of chat GBT:
![image](https://github.com/MicroStrategy/rest-api-docs/assets/139224196/c6633391-c8b9-4c69-92b6-63497746a6a1)

What is the type parameter for a Dossier please? Report being 3.

**Comment by ChristiaanEmpowerBI** [comment]
I was scrolling through this site:
https://www2.microstrategy.com/producthelp/Current/ReferenceFiles/reference/constant-values.html#com.microstrategy.webapi.EnumDSSXMLObjectTypes.DssXmlTypeUnknown

And by sheer luck tried **55**. It is the same for Dossiers and Documents. 

Now that I have the folder structure, is there a way to retrieve the last time a report or dossier was opened? I only found the create date and the last modified date.

**Reply by xiakunhou** [reply]
Yes all types can be found in https://www2.microstrategy.com/producthelp/Current/ReferenceFiles/reference/com/microstrategy/webapi/EnumDSSXMLObjectTypes.html

Last access time, have not seen this before.

**Comment by xiakunhou** [comment, marked as answer]
For searching objects, you can use quick search API.

https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Browsing/doQuickSearch

**Reply by ChristiaanEmpowerBI** [reply]
Thanks I looked into this API and it will solve my requests, apart from the last accessed date. There I think you will need to query the Enterprise Manager or Platform Analytics projects and compare the results.
